theano/tensor/opt.py

```diff
@@ -1430,19 +1436,29 @@ def same_shape(self, x, y, dim_x=None, dim_y=None):
 
 class ShapeOptimizer(Optimizer):
     """Optimizer that serves to add ShapeFeature as an fgraph feature."""
-    def __init__(self):
-        Optimizer.__init__(self)
-
     def add_requirements(self, fgraph):
         fgraph.attach_feature(ShapeFeature())
 
     def apply(self, fgraph):
         pass
 
+
+class UnShapeOptimizer(Optimizer):
+    """Optimizer remove ShapeFeature as an fgraph feature."""
+    def apply(self, fgraph):
+        for feature in fgraph._features:
+            if isinstance(feature, ShapeFeature):
+                fgraph.remove_feature(feature)
```

Comment:
I don't understand why it would invoke old one. There can be only 1 shape feature per fgraph. ShapeFeature.on_attach assert that.